python-stubs/dolphin/_modules/event.pyi

```diff
@@ -29,63 +24,48 @@ class _MemorybreakpointCallback(Protocol):
         :param value: new value at the given address
         """
 
-
-def on_memorybreakpoint(callback: Optional[_MemorybreakpointCallback]):
+def on_memorybreakpoint(callback: _MemorybreakpointCallback | None) -> None:
     """
     Registers a callback to be called every time a previously added memory breakpoint is hit.
 
     :param callback:
     :return:
     """
 
+async def memorybreakpoint() -> tuple[bool, int, int]:
+    """Awaitable event that completes once a previously added memory breakpoint is hit."""
 
-async def memorybreakpoint() -> (bool, int, int):
-    """
-    Awaitable event that completes once a previously added memory breakpoint is hit.
-    """
-
-
+@type_check_only
 class _SaveStateCallback(Protocol):
     def __call__(self, isSlot: bool, slot: int) -> None:
```

Comment:
By convention this param should be `is_slot`. But that's not something I can change in eth stubs without also having to change in the source (unless it's a positional-only argument)

Reply:
From what I can tell, the names are pulled from the structs seen here: https://github.com/TASLabz/dolphin/blob/master/Source/Core/Core/API/Events.h#L45-L54

Reply:
If that's the case then I guess this should be a positional-only callback-protocol
```suggestion
    def __call__(self, is_slot: bool, slot: int, /) -> None:
```